x/commitment/types/codec.go

```diff
@@ -0,0 +1,47 @@
+package types
+
+import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/msgservice"
+)
+
+func RegisterCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&MsgCommitTokens{}, "commitment/CommitTokens", nil)
+	cdc.RegisterConcrete(&MsgUncommitTokens{}, "commitment/UncommitTokens", nil)
+	cdc.RegisterConcrete(&MsgWithdrawTokens{}, "commitment/WithdrawTokens", nil)
+	cdc.RegisterConcrete(&MsgDepositTokens{}, "commitment/DepositTokens", nil)
+	cdc.RegisterConcrete(&MsgVest{}, "commitment/Vest", nil)
+	cdc.RegisterConcrete(&MsgCancelVest{}, "commitment/CancelVest", nil)
+	// this line is used by starport scaffolding # 2
+}
+
+func RegisterInterfaces(registry cdctypes.InterfaceRegistry) {
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgCommitTokens{},
+	)
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgUncommitTokens{},
+	)
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgWithdrawTokens{},
+	)
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgDepositTokens{},
+	)
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgVest{},
+	)
+	registry.RegisterImplementations((*sdk.Msg)(nil),
+		&MsgCancelVest{},
+	)
```

Comment:
I think it would be good to combine these registrations into a single function
```
registry.RegisterImplementations((*sdk.Msg)(nil),
		&MsgCommitTokens{},
		&MsgUncommitTokens{},
		&MsgWithdrawTokens{},
		&MsgDepositTokens{},
		&MsgVest{},
		&MsgCancelVest{},
	)
```